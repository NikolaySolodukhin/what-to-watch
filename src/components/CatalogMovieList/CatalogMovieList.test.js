import React from 'react';
import renderer from 'react-test-renderer';
import CatalogMoviesList from './CatalogMovieList';

describe(`CatalogMoviesList`, () => {
  it(`CatalogMoviesList renders correctly`, () => {
    const mockProps = {
      filmsList: [{
        id: 1,
        genreId: 1,
        backgroundColor: `white`,
        backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        description: `Film`,
        director: `Raccoon`,
        genre: `comedian`,
        isFavorite: false,
        name: `Fantastic Beasts`,
        posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        previewVideoLink: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        rating: 4,
        released: 2018,
        runTime: 1990,
        scoresCount: 10,
        starring: [],
        videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
      }],
    };
    const tree = renderer
      .create(<CatalogMoviesList filmsList={mockProps.filmsList} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
