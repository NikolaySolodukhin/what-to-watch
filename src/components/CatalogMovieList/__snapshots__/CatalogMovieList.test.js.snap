// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogMoviesList CatalogMoviesList renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/1"
        onClick={[Function]}
      >
        Fantastic Beasts
      </a>
    </h3>
  </article>
</div>
`;
